src/middleware/authorisation.js

```diff
@@ -49,6 +49,9 @@ export async function authorise (ctx, done) {
     // authorisation succeeded
     ctx.authorisedChannel = channel
     logger.info(`The request, '${ctx.request.path}' is authorised to access ${ctx.authorisedChannel.name}`)
+  } else if(!channel) {
```

Comment:
@rcrichton 
What are your thoughts on auth here.

Returning 401 appears to be revealing the existence of a channel to unauthenticated users? I think this is a problem.
Returning 404 in `no channel` and `auth failed` would mean "No channel exists for your credentials" - which protects the existence of channels from unauthenticated users.
Or is this overboard.

Reply:
Hmm, they would have to have guessed the path already, so I don't think this is too big a deal. It might also make issues harder to debug for a user. My feeling is that we leave it as is.